hazelcast/src/test/java/com/hazelcast/client/map/ClientMapQueryIterableTest.java

```diff
@@ -0,0 +1,236 @@
+            String key = generateKeyForPartition(server, 1);
+            intMap.put(key, i);
+        }
+
+        final Iterator<Entry<String, Integer>> iterator =
+                intMap.iterable(10, Projections.identity(), new EvenPredicate()).iterator();
+
+        final ArrayList<Entry<String, Integer>> projected = collectAll(iterator);
+        for (Entry<String, Integer> i : projected) {
+            assertEquals(0, i.getValue() % 2);
+        }
+
+        final Collection<Integer> actualValues = intMap.values();
+        assertEquals(actualValues.size() / 2, projected.size());
+
+
+        for (Entry<String, Integer> e : intMap.entrySet()) {
+            if (e.getValue() % 2 == 0) {
+                assertTrue(projected.contains(e));
+            }
+        }
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void test_remove_Throws_Exception() {
+        final ClientMapProxy<String, String> proxy = getMapProxy();
+        final Iterator<String> iterator = proxy.iterable(10, new TestProjection(), Predicates.alwaysTrue()).iterator();
+
+        iterator.remove();
+    }
+
+    @Test
+    public void test_Next_Returns_Values_When_FetchSizeExceeds_On_NonEmptyPartition() {
+        final ClientMapProxy<String, String> proxy = getMapProxy();
+        String value = randomString();
+        for (int i = 0; i < 100; i++) {
+            String key = generateKeyForPartition(server, 1);
+            proxy.put(key, value);
+        }
+        final Iterator<String> iterator = proxy.iterable(10, new GetValueProjection<>(), Predicates.alwaysTrue()).iterator();
+        for (int i = 0; i < 100; i++) {
+            String val = iterator.next();
+            assertEquals(value, val);
+        }
+    }
+
+    private <T> ArrayList<T> collectAll(Iterator<T> iterator) {
+        final ArrayList<T> projected = new ArrayList<>();
+        while (iterator.hasNext()) {
+            projected.add(iterator.next());
+        }
+        return projected;
+    }
+
+    protected ClientConfig getClientConfig() {
+        return new ClientConfig();
+    }
+
+    private <K, V> ClientMapProxy<K, V> getMapProxy() {
+        String mapName = randomString();
+        return (ClientMapProxy<K, V>) client.getMap(mapName);
+    }
+
+    private static class EvenPredicate implements Predicate<String, Integer> {
+        @Override
+        public boolean apply(Entry<String, Integer> mapEntry) {
+            return mapEntry.getValue() % 2 == 0;
+        }
+    }
+
+    private static class TestProjection implements Projection<Entry<String, String>, String> {
+        @Override
+        public String transform(Entry<String, String> input) {
+            return "dummy" + input.getValue();
+        }
+    }
+
+    private static class GetValueProjection<T> implements Projection<Entry<String, T>, T> {
+        @Override
+        public T transform(Entry<String, T> input) {
+            return input.getValue();
+        }
+    }
+}
```

Comment:
Do we test the public methods somewhere? I mean the `map.iterator()` and `map.iterator(fetchSize)`